multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ExtractServicesDeploymentStep.java

```diff
@@ -0,0 +1,55 @@
+package org.cloudfoundry.multiapps.controller.process.steps;
+
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudServiceInstanceExtended;
+import org.cloudfoundry.multiapps.controller.core.cf.CloudHandlerFactory;
+import org.cloudfoundry.multiapps.controller.core.cf.v2.ServicesCloudModelBuilder;
+import org.cloudfoundry.multiapps.controller.core.security.serialization.SecureSerialization;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import javax.inject.Named;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Named("extractServicesDeploymentStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class ExtractServicesDeploymentStep extends SyncFlowableStep {
+
+    @Override
+    protected StepPhase executeStep(ProcessContext context) {
+        getStepLogger().debug(Messages.EXTRACT_SERVICES_FROM_BATCH);
+        // get singular batch of resources variable
+        List<Resource> batchOfResources = context.getVariable(Variables.BATCH_TO_PROCESS);
+        // extract all services from the batch of resources
+        ServicesCloudModelBuilder servicesCloudModelBuilder = getServicesCloudModelBuilder(context);
+        List<CloudServiceInstanceExtended> servicesCalculatedForDeployment = servicesCloudModelBuilder.build(batchOfResources);
+        // Build a list of services for creation and save them in the context:
```

Comment:
I'm not a fan of comments, and I think the code describes itself enough, I'd suggest to remove them